### PR TITLE
Changes needed to make kl/sync-layer-reorg server compatible 

### PR DESCRIPTION
**Pull request by StanislavBreadless**
# What ❔

## Checklist

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.

### CODE REVIEW

File: l2-contracts/contracts/bridge/L2AssetRouter.sol [hunk 1]
```diff
@@ -29,16 +29,17 @@ contract L2AssetRouter is IL2AssetRouter, Initializable {
     /// @dev The address of the L2 legacy shared bridge.
     address public immutable L2_LEGACY_SHARED_BRIDGE;
 
-    /// @dev The address of the L1 asset router counterpart.
-    address public override l1AssetRouter;
```
Comment (kelemeno) [inline l2-contracts/contracts/bridge/L2AssetRouter.sol:33]: Why rename it? The L2AssetRouter is not the L2SharedBrideLegacy, it will be at a different address, we don't need storage and interface backward compatibility. Same question for some of the functions, e.g. withdraw

File: l2-contracts/contracts/bridge/L2NativeTokenVault.sol [hunk 2]
```diff
@@ -188,6 +188,9 @@ contract L2NativeTokenVault is IL2NativeTokenVault, Ownable2StepUpgradeable {
     function l2TokenAddress(address _l1Token) public view override returns (address expectedToken) {
         bytes32 expectedAssetId = DataEncoding.encodeNTVAssetId(L1_CHAIN_ID, _l1Token);
         expectedToken = tokenAddress[expectedAssetId];
+        if (expectedToken == address(0)) {
```
Comment (kelemeno) [inline l2-contracts/contracts/bridge/L2NativeTokenVault.sol:191]: I wanted to remove this. The l2TokenAddress is backwards compatible on the L2SharedBridgeLegacy, but on the L2NTV I only want to return the deployed address, otherwise the user should call the calculateCreate2TokenAddress

File: l1-contracts/contracts/bridgehub/Bridgehub.sol [hunk 3]
```diff
@@ -286,7 +286,6 @@ contract Bridgehub is IBridgehub, ReentrancyGuard, Ownable2StepUpgradeable, Paus
         bytes calldata _initData,
         bytes[] calldata _factoryDeps
     ) external onlyOwnerOrAdmin nonReentrant whenNotPaused onlyL1 returns (uint256) {
-        require(L1_CHAIN_ID == block.chainid, "BH: New chain registration only allowed on L1");
```
Comment (kelemeno) [inline l1-contracts/contracts/bridgehub/Bridgehub.sol:289]: why was this removed?

Comment (StanislavBreadless) [reply l1-contracts/contracts/bridgehub/Bridgehub.sol:289]: because `onlyL1` is already present

**Review by kelemeno — APPROVED** [review]
